app/src/test/java/ingsw/group1/findmyphone/contacts/SMSContactConverterTest.java

```diff
@@ -0,0 +1,60 @@
+package ingsw.group1.findmyphone.contacts;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import ingsw.group1.msglibrary.SMSPeer;
+
+/**
+ * Test for {@link SMSContactConverterUtils}
+ *
+ * @author Giorgia Bortoletti
+ */
+public class SMSContactConverterTest {
+
+    private static final String EX_VALID_ADDRESS = "+393478989890"; //for contact and peer
+    private static final String CONTACT_VALID_NAME = "NewContact";
+    private static final String CONTACT_EMPTY_NAME = "";
+
+    private SMSContact contactWithoutName;
+    private SMSContact contactWithName; //a contact may or may not have a name
+    private SMSPeer peerTest;
+
+    @Before
+    public void createPeerAndContact(){
+        contactWithoutName = new SMSContact(EX_VALID_ADDRESS, CONTACT_EMPTY_NAME);
+        contactWithName = new SMSContact(EX_VALID_ADDRESS, CONTACT_VALID_NAME);
+        peerTest = new SMSPeer(EX_VALID_ADDRESS);
+    }
+
+    /**
+     * Test {@link SMSContactConverterUtils#contactFromSMSPeer(SMSPeer)}
+     *
+     * Conversion from a SMSPeer to a Contact without a name
+     */
+    @Test
+    public void contactFromSMSPeerWithEmptyName() {
+        Assert.assertEquals(contactWithoutName, SMSContactConverterUtils.contactFromSMSPeer(peerTest));
+    }
+
+    /**
+     * Test {@link SMSContactConverterUtils#contactFromSMSPeer(SMSPeer, String)}
+     *
+     * Conversion from a SMSPeer to a Contact with a name
+     */
+    @Test
+    public void contactFromSMSPeerWithValidName() {
+        Assert.assertEquals(contactWithName, SMSContactConverterUtils.contactFromSMSPeer(peerTest, CONTACT_VALID_NAME));
+    }
+
+    /**
+     * Test {@link SMSContactConverterUtils#peerFromContact(SMSContact)}
+     */
+    @Test
+    public void peerFromContact() {
+        Assert.assertEquals(peerTest, SMSContactConverterUtils.peerFromContact(contactWithoutName));
```

Comment:
Weird, Codacy should be pissed by the double assertion, whatever, not a huge deal.

Reply:
What do you mean for assertion? 'Assert.assertEquals' ?

Reply:
Yes, usually tests should have a single assert/fail statement. You can always combine the two statements into a single `assertTrue` or `assertFalse`, and Codacy usually warns about this, but this time it didn't, no clue why

Reply:
I don't know why